instrumentation/rpc/src/main/java/brave/rpc/RpcClientHandler.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright 2013-2020 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package brave.rpc;
+
+import brave.Span;
+import brave.SpanCustomizer;
+import brave.Tracer;
+import brave.internal.Nullable;
+import brave.propagation.TraceContext;
+import brave.propagation.TraceContext.Injector;
+import brave.sampler.SamplerFunction;
+
+/**
+ * This standardizes a way to instrument rpc clients, particularly in a way that encourages use of
+ * portable customizations via {@link RpcRequestParser} and {@link RpcResponseParser}.
+ *
+ * <p>Synchronous interception is the most straight forward instrumentation.
+ *
+ * <p>You generally need to:
+ * <ol>
+ *   <li>Start the span and add trace headers to the request</li>
+ *   <li>Put the span in scope so things like log integration works</li>
+ *   <li>Invoke the request</li>
+ *   <li>Catch any errors</li>
```

Comment:
I'd say `Catch any errors, tag them and throw them up`

Reply:
catch is maybe imprecise in hindsight as sometimes they are just passed to us. Ack the example catch and rethrows in sync code, but probably we shouldn't focus on the catching part.

`If there was a Throwable, add it to the span`

PS better english for what you said would be

"Catch, tag, and rethrow any errors" because "throw them up" is saying more like vomit, which well.. isn't wrong :)